syntax = "proto3";
import "google/protobuf/empty.proto";
option go_package = ".;proto";

service Address {
    rpc GetAddressList(AddressRequest) returns (AddressListResponse);
    rpc CreateAddress(AddressRequest) returns (AddressResponse);
    rpc UpdateAddress(AddressRequest) returns (google.protobuf.Empty);
    rpc DeleteAddress(AddressRequest) returns (google.protobuf.Empty);
}

message AddressRequest {
    int32 id = 1;
    int32 userId = 2;
    string province = 3;
    string city = 4;
    string district = 5;
    string address = 6;
    string signerName = 7;
    string signerMobile = 8;
}

message AddressResponse {
    int32 id = 1;
    int32 userId = 2;
    string province = 3;
    string city = 4;
    string district = 5;
    string address = 6;
    string signerName = 7;
    string signerMobile = 8;
}

message AddressListResponse {
    int32 total = 1;
    repeated AddressResponse data = 2;
}
